frontend/public/components/modals/remove-volume-modal.tsx

```diff
@@ -0,0 +1,90 @@
+import * as _ from 'lodash-es';
+import * as React from 'react';
+
+import { createModalLauncher, ModalTitle, ModalBody, ModalSubmitFooter } from '../factory';
+import { history, resourcePathFromModel } from '../utils';
+import { getVolumeType, K8sKind, k8sPatch, K8sResourceKind } from '../../module/k8s/';
+import { RowVolumeData } from '../volumes-table';
+
+export const RemoveVolumeModal: React.FC<RemoveVolumeModalProps> = (props) => {
+  const [inProgress, setInProgress] = React.useState(false);
+  const [errorMessage, setErrorMessage] = React.useState('');
+
+  const getRemoveVolumePatch = (resource, rowVolumeData: RowVolumeData) => {
+    const location = resource.kind === 'Pod' ? 'spec' : 'spec.template.spec';
+    const dirLocation = resource.kind === 'Pod' ? '/spec' : '/spec/template/spec';
+    const containers = _.get(resource, `${location}.containers`, []);
+    const patches = [];
+    let mountCount = 0;
+    containers.map((container, i) => {
+      const mounts = _.get(container, 'volumeMounts', []);
+      mounts.map((mount, j) => {
+        if (mount.name === rowVolumeData.name) {
+          mountCount++;
+          if (mount.mountPath === rowVolumeData.mountPath) {
+            patches.push({op: 'remove', path: `${dirLocation}/containers/${i}/volumeMounts/${j}`});
+            mountCount--;
```

Comment:
Instead of incrementing and decrementing, just increment in an `else` clause

Reply:
done